src/main/java/com/aviva/CarRecommendations/SensoRate.java

```diff
@@ -10,44 +11,41 @@
+        float carPrice = car.getPrice();
+        float downPayment = (float) (0.1 * carPrice);
+        float vehicleKMS = 15000.0F; // Assuming average kilometres driven is 15k
+        String carMake = car.getMake();
+        String carModel = car.getModel();
+        int carYear = car.getYear();
+        float loanAmount = (float) (carPrice - downPayment + (0.2 * carPrice)); // Adding 20% of car value as upselling
 
         // Input parameters for the Senso API's rate endpoint
-        String inputjson = "{\n" +
-                "  \"loanAmount\": "+ user.getSavings() +",\n" +
-                "  \"creditScore\": "+ user.getCreditScore() +",\n" +
-                "  \"pytBudget\": "+ user.getMonthlyBudget() +",\n" +
-                "  \"vehicleMake\": \"Honda\",\n" +
-                "  \"vehicleModel\": \"Civic\",\n" +
-                "  \"vehicleYear\": 2021,\n" +
-                "  \"vehicleKms\": 1000\n" +
-                "}";
-
-        // Creates a POST request
+        String inputJson = String.format("{\"loanAmount\": %f, \"creditScore\": %d, \"pytBudget\": %d," +
+                " \"vehicleMake\": \"%s\", \"vehicleModel\": \"%s\", \"vehicleYear\": %d, \"vehicleKms\": %f," +
+                " \"listPrice\": %f, \"downpayment\": %f}", loanAmount, user.getCreditScore(), (int) user.getMonthlyBudget(),
+                carMake, carModel, carYear, vehicleKMS, carPrice, downPayment);
+
+         // Creates a POST request
         HttpRequest request = HttpRequest.newBuilder()
-                .uri(URI.create(System.getenv("SENSO_API_URL")))
+                .uri(URI.create(System.getenv("SENSO_API_URL") + "/rate"))
                 .header("Content-Type", "application/json")
                 .header("x-api-key", System.getenv("SENSO_API_KEY"))
-                .POST(HttpRequest.BodyPublishers.ofString(inputjson))
+                .POST(HttpRequest.BodyPublishers.ofString(inputJson))
                 .build();
 
-        // use the client to send the request
-        HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
-
-        //parse the returned JSON file
-        return getLoanAmountFromJSON(response.body());
-    }
-
-    private static float getLoanAmountFromJSON(String response) {
-        JSONObject obj = new JSONObject(response);
-        return obj.getFloat("amount");
-    }
-
-    public static void main(String[] args) throws IOException, InterruptedException {
-        AccountHolder user;
-        AccountHolderDataInterface accountData = new CSVAccountHolderData();
-        user = accountData.getClientByID("5002357538983918");
-        System.out.println(getApprovedLoanAmount(user));
+        try {
+            // use the client to send the request
+            HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
+            JSONObject responseJSON = new JSONObject(response.body());
+            return responseJSON.getFloat("interestRate");
```

Comment:
We need more than just the interest rate from Senso

Reply:
if you can save the installment information, or at least how much we are paying for that car per month

Reply:
we have made method to get a car's installments so you can use this however need be